conans/model/build_info.py

```diff
@@ -421,6 +440,9 @@ def get_sorted_components(self):
                 req_processed = [n for n in c.required_component_names if n not in processed]
                 if not req_processed and name not in processed:
                     processed.append(name)
+            if not processed:
```

Comment:
I think this might not cover all loops. This might need an aggregated variable in the internal ``for`` loop, that checks that processed is actually growing in every iteration.

Reply:
I have been checking the loop and, from my point of view, there is no evidence that we need to have another aggregated variable. Do you have any idea about a possible situation that it could fail?

Reply:
Added one more test to check another kind of cycle